util/stringutil/string_util.go

```diff
@@ -234,3 +235,8 @@ func DoMatch(str string, patChars, patTypes []byte) bool {
 	}
 	return sIdx == len(str)
 }
+
+// CopyString deep copies a string.
+func CopyString(src string) string {
```

Comment:
How about `s/CopyString/Copy/`? We usually call the function with the package name, "stringutil.Copy()" would be simpler.